Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -386,128 +436,71 @@ namespace AzToolsFramework
-                for (int i = 0; i < NumOfDefaultLocationsInLevel; i++)
-                {
-                    AZStd::string finalPath =
-                        "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
-                    registry->SetObject(finalPath, ViewBookmark());
-                }
-
-                LoadViewBookmarks();
-                return true;
-            }
-            else
-            {
-                LoadViewBookmarks();
-
-                for (int i = 0; i < m_localBookmarks.size(); i++)
-                {
-                    AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(i);
-                    registry->SetObject(finalPath, m_localBookmarks.at(i));
-                }
-
-                return true;
-            }
-        }
-        else // if the field is empty we don't have a file linked to the prefab, so we create one and we save it in the
-             // component
+        // if the field is empty, we don't have a file linked to the prefab, so we create one and we save it in the component
+        if (bookmarkComponent->GetLocalBookmarksFileName().empty())
         {
-            m_bookmarkfileName = GenerateBookmarkFileName();
-
-            // Initialize default locations to 0
-            for (int i = 0; i < NumOfDefaultLocationsInLevel; i++)
-            {
-                AZStd::string finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
-                registry->SetObject(finalPath, ViewBookmark());
-            }
-
-            LoadViewBookmarks();
-            bookmarkComponent->SetLocalBookmarksFileName(m_bookmarkfileName);
-            return true;
+            bookmarkComponent->SetLocalBookmarksFileName(GenerateBookmarkFileName());
         }
+
+        m_bookmarkFileName = bookmarkComponent->GetLocalBookmarksFileName();
     }
 
     bool LocalViewBookmarkLoader::SaveLocalBookmark(const ViewBookmark& bookmark, ViewBookmarkType bookmarkType)
     {
-        LoadDefaultLocalViewBookmarks();
-
-        AZStd::string finalPath;
+        SetupLocalViewBookmarkComponent();
 
+        AZStd::string setRegPath;
         switch (bookmarkType)
         {
         case ViewBookmarkType::Standard:
-            finalPath = "/" + m_bookmarkfileName + "/LocalBookmarks/" + AZStd::to_string(m_localBookmarkCount++);
+            // note: this function is not used currently and will need to be updated when DefaultViewBookmarkCount is removed
+            setRegPath = BookmarkSetRegPath(m_bookmarkFileName, aznumeric_cast<int>(m_localBookmarks.size()) + 1);
             break;
         case ViewBookmarkType::LastKnownLocation:
-            finalPath = "/" + m_bookmarkfileName + "/LastKnownLocation";
+            setRegPath = "/" + m_bookmarkFileName + "/LastKnownLocation";
             break;
         }
```

Comment:
This switch could have a default case with a warning.

Reply:
Good call, please see this PR for the update https://github.com/o3de/o3de/pull/8434 @moraaar